experimental/Permutations/Permutations.jl

```diff
@@ -0,0 +1,136 @@
+# This file implements a new way to input permutations in Julia. For example
+# it is possible to create a permutation as follow
+# pi = Oscar.Permutations.@perm (1,2,3)(4,5)(6,7,8)
+# > (1,2,3)(4,5)(6,7,8)
+# For this we use macros to modify the syntax tree of (1,2,3)(4,5)(6,7,8) such that
+# Julia can deal with the expression.
+
+module Permutations
+
+ using GAP
+ using Oscar
+
+################################################################################
+# Macro to input a permutation as
+# julia> pi = Oscar.Permutations.@perm (1,2,3)(4,5)(6,7,8)
+# > (1,2,3)(4,5)(6,7,8)
+#
+macro perm(ex)
+    h = (ex).head
+    arg = (ex).args
+    res = []
```

Comment:
```suggestion
    res = Expr[]
```
for type stability. Might not really matter here, but it's a good habit to learn about this sooner rather than later: if you write `[]` it will be a `Vector{Any}`, with my change a `Vector{Expr}` and that then allows for more optimization opportunities later on.

Reply:
This throws an error
'''
ERROR: LoadError: TypeError: in Expr, expected Symbol, got a value of type Vector{Any}
'''

Reply:
OK, maybe I got the type wrong then. You could insert something like `println(typeof(res))` at the end of the macro to see what it typically is like.

Anyway, I think you should just forget about it, it's not worth it: this code is only executed during precompilation or in a REPL, and a few microseconds here or there won't matter